beat/apps/importFaultShp.py

```diff
@@ -0,0 +1,239 @@
+    
+    def addStrikes(self,data):
+        self.segmentStrike.append(data)
+    
+    def addLengths(self,data):
+        self.segmentLength.append(data)
+    
+    def plotFault(self):
+        from matplotlib import pyplot as plt
+        
+        #plot fault segments
+        z=1
+        for i in range(len(self.segmentPointsXY)):
+            for j in range(len(self.segmentPointsXY[i])):
+                plt.plot(self.segmentPointsXY[i][j,:].reshape(2,2)[0],self.segmentPointsXY[i][j,:].reshape(2,2)[1],'-k')
+                plt.plot(self.segmentMidPoints[i][j,0],self.segmentMidPoints[i][j,1],'bo')
+                plt.text(self.segmentMidPoints[i][j,0]+100,self.segmentMidPoints[i][j,1]+100, 'F{} , Strike: '.format(z) + str(int(self.segmentStrike[i][j]))+'°')
+                z = z +1 
+        #plot event location
+        plt.plot(0,0,'r+', label = 'Event location',ms=14)
+        plt.xlabel("East shift [m]")
+        plt.ylabel("North shift [m]")
+        plt.title(" Fault segements to model in BEAT")
+        
+
+def import_faultShp(daShapefile):
+    try:
+        from osgeo import ogr
+        print('Import of ogr from osgeo worked!\n')
+    except:
+        print('Import of ogr from osgeo failed\n\n')
+    
+    # import data from shapefile
+    driver = ogr.GetDriverByName('ESRI Shapefile')
+    data = driver.Open(daShapefile,0)
+    
+    # Check to see if shapefile is found.
+    if data is None:
+        print('Could not open %s' % daShapefile)
+    else:
+        print('Opened %s' % daShapefile)
+        layer = data.GetLayer()
+        featureCount = layer.GetFeatureCount()
+        print("Number of features in %s: %d" % (os.path.basename(daShapefile),featureCount))
+        
+    #extract Fault Point Coordinates from shapefile
+    faultPoints = {}
+    
+    for i in range(featureCount):
+        feature         = layer.GetFeature(i)
+        featureGeometry = feature.GetGeometryRef()
+        fnPoints        = featureGeometry.GetPointCount()
+        featurePoints   = np.zeros((fnPoints,2))
+        
+        #extract points from shapefile features
+        for j in range(fnPoints): 
+            featurePoints[j,:] = np.asarray(featureGeometry.GetPoint(j)[0:2])
+            
+        faultPoints[i] = featurePoints
+        
+    # Output array of fault points
+    return faultPoints
+
+
+def getFaultSegments(event, faultSource):
+    from pyrocko import orthodrome
+    #Get fault points coordinates in local coordinate system with respect to the event location
+    
+    # Get point coordinates from shapefile 
+    # event :: list [lon, lat]
+    # faultSource :: str "filename.shp"
+    fault = import_faultShp(faultSource)
+    
+    fSegments = faultSegments()
+    
+    #get Points from fault features - lines
+    for i in range(len(fault)):
+        faultXY = orthodrome.latlon_to_ne_numpy(fault[i][:,1],fault[i][:,0],event[1],event[0]) 
+        
+        fSegmentPointsLonLat = np.zeros((len(faultXY[0])-1,4))
```

Comment:
faultXY is a numpy array, so instead of len(faultXY[0]), its better to use:
faultXY.shape[0]
why the -1? Because of the closed polygons thus need to delete last line?
define nsegment_points = faultXY.shape[0] and then reuse instead of reevaluating many times.